web/modules/custom/joinup_core/src/EventSubscriber/RefreshCachedFieldsEventSubscriber.php

```diff
@@ -355,4 +317,48 @@ protected function getPiwikSettings($bundle) {
     return FALSE;
   }
 
+  /**
+   * Builds a list of parameters for a sub query.
+   *
+   * This method gathers all information needed for each sub query separately.
+   * All the default parameters, e.g. the authentication token and the site id,
+   * are set automatically by generating a new query each time.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity with the expired field.
+   * @param array $parameters
+   *   A list of extra parameters to pass to the array of parameters.
+   *
+   * @return array
+   *   A list of parameters.
+   */
+  protected function getSubQueryParameters(EntityInterface $entity, array $parameters = []): array {
+    $bundle = $entity->bundle();
+    $period = $this->getTimePeriod($bundle);
+    $type = $this->getType($bundle);
+    $method = $this->getPiwikMethod($bundle);
+    $url_parameter_name = $this->getUrlParameterName($bundle);
+    $url_parameter = $this->getUrlParameter($entity);
+
+    $sub_query = $this->piwikQueryFactory->getQuery($method);
+    $date_range = [
+      // If the period is 0 we should get all results since launch.
+      $period > 0 ? (new DateTimePlus("$period days ago"))->format('Y-m-d') : $this->configFactory->get('joinup_core.piwik_settings')->get('launch_date'),
+      (new DateTimePlus())->format('Y-m-d'),
+    ];
+
+    $parameters['period'] = 'year';
```

Comment:
When using the year range, this results in requesting the total of 'views' for the requested years.
This could lead to a weird sliding window, depending on the point in the year.
A solution could be to always request the previous and the current year.
That way at least the proportions won't be too much off.